### ISSUE TITLE
Powered OFF worker node should not be considered during upgrade

### ISSUE THREAD

**Issue by soumyalj** [rancher/rke#1907]
**RKE version:** v1.1.0-rc5

**Docker version: (`docker version`,`docker info` preferred)** 19.03

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)** Ubuntu

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**EC2

**cluster.yml file:**

**Steps to Reproduce:**
1. Create a RKE Cluster with 3 control planes, 1 etcd node and 5 worker nodes(NW1 to NW5)
```
ssh_key_path: /Users/soumya/.ssh/jenkins-rke-validation.pem
auth:
  strategy: x509
nodes:
  - address: abc
    user: ubuntu
    role: [etcd]
  - address: def
    user: ubuntu
    role: [controlplane]
  - address: ghk
    user: ubuntu
    role: [controlplane]
  - address: ijk
    user: ubuntu
    role: [controlplane]
  - address: nlp
    user: ubuntu
    role: [worker]
  - address: abcd
    user: ubuntu
    role: [worker]
  - address: efgh
    user: ubuntu
    role: [worker]
  - address: sdfds
    user: ubuntu
    role: [worker]
  - address: dsfsdfs
    user: ubuntu
    role: [worker]
#kubernetes_version: v1.17.2-rancher1-1
kubernetes_version: v1.16.6-rancher1-1
```
2. Create pods and daemonsets in the cluster
3. Power off one worker node ex: NW5
4. Run RKE up again by changing the kubelet extra args
```
ssh_key_path: /Users/soumya/.ssh/jenkins-rke-validation.pem
auth:
  strategy: x509
nodes:
  - address: abc
    user: ubuntu
    role: [etcd]
  - address: def
    user: ubuntu
    role: [controlplane]
  - address: ghk
    user: ubuntu
    role: [controlplane]
  - address: ijk
    user: ubuntu
    role: [controlplane]
  - address: nlp
    user: ubuntu
    role: [worker]
  - address: abcd
    user: ubuntu
    role: [worker]
  - address: efgh
    user: ubuntu
    role: [worker]
  - address: sdfds
    user: ubuntu
    role: [worker]
  - address: dsfsdfs
    user: ubuntu
    role: [worker]

#kubernetes_version: v1.17.2-rancher1-1
kubernetes_version: v1.16.6-rancher1-1

upgrade_strategy:
 max_unavailable: 3
 upgrade_strategy:
  drain: true
  node_drain_input:
    force: false
    ignore_daemonsets: true
    delete_local_data: false
    grace_period: -1
    timeout: 60

services:
   kubelet:
     extra_args:
       max-pods: 145



```

rke up fails with the below error where the powered OFF node is still considered for the upgrade.

```
INFO[0044] [controlplane] Processing control plane components for upgrade one at a time 
INFO[0044] Processing controlplane host abcdef    
INFO[0044] [authz] Creating rke-job-deployer ServiceAccount 
INFO[0045] [authz] rke-job-deployer ServiceAccount created successfully 
INFO[0045] [authz] Creating system:node ClusterRoleBinding 
INFO[0045] [authz] system:node ClusterRoleBinding created successfully 
INFO[0045] [authz] Creating kube-apiserver proxy ClusterRole and ClusterRoleBinding 
INFO[0045] [authz] kube-apiserver proxy ClusterRole and ClusterRoleBinding created successfully 
INFO[0045] Successfully Deployed state file at [./clusterzeroworker.rkestate] 
INFO[0045] [state] Saving full cluster state to Kubernetes 
INFO[0046] [state] Successfully Saved full cluster state to Kubernetes ConfigMap: cluster-state 
INFO[0046] [worker] Upgrading Worker Plane..            
INFO[0046] 3 worker nodes can be unavailable at a time  
INFO[0047] First checking and processing worker components for upgrades on nodes with etcd/controlplane roles one at a time 
INFO[0047] [workerplane] Processing host abdcdfd    
ERRO[0047] Failed to upgrade hosts:NW5 with error [host NW5 not ready] 
FATA[0047] [workerPlane] Failed to upgrade Worker Plane: [host NW5 not ready] 


```
powered OFF node is still considered for the upgrade.

**Comment by mrajashree** [comment]
Similar to https://github.com/rancher/rke/issues/1909
Also the drain input format should be:
```
// default values
upgrade_strategy:
  max_unavailable: 3
  drain: true
  node_drain_input:
    force: false
    ignore_daemonsets: true
    delete_local_data: false
    grace_period: -1
    timeout: 60
```

**Comment by mrajashree** [comment]
Available with v1.1.0-rc7

**Comment by soumyalj** [comment]
Tested with RKE version v1.1.0-rc7

1. Create a RKE Cluster with 3 control planes, 1 etcd node and 5 worker nodes(NW1 to NW5) with k8s version v1.16.7-rancher1-1
2. Power OFF one worker node NW2
3. Upgrade to a new K8s version v1.17.3-rancher1-1 using the below upgrade strategy

```
kubernetes_version: v1.17.3-rancher1-1

upgrade_strategy:
  max_unavailable: 2
  drain: true
  node_drain_input:
     force: false
     ignore_daemonsets: true
     delete_local_data: false
     grace_period: -1
     timeout: 120
```

Upgrade continues further without any error. Worker nodes other than the powered OFF node upgrade successfully. Powered OFF node is skipped as expected. Logs provide a message that the powered OFF node was skipped as below:

```
INFO[0332] [ingress] Setting up nginx ingress controller 
INFO[0332] [addons] Saving ConfigMap for addon rke-ingress-controller to Kubernetes 
INFO[0332] [addons] Successfully saved ConfigMap for addon rke-ingress-controller to Kubernetes 
INFO[0332] [addons] Executing deploy job rke-ingress-controller 
INFO[0333] [ingress] ingress controller nginx deployed successfully 
INFO[0333] [addons] Setting up user addons              
INFO[0333] [addons] no user addons defined              
FATA[0333] Provisioning incomplete, host(s) [abcd] skipped because they could not be contacted 

```
Etcd and control planes got upgraded correctly. New pods were created successfully in the cluster.
